### PR TITLE
chore: prevent action bash escapes

### PR DESCRIPTION
**Pull request by rbtr**
# Description

@robertprast in #285 and @random-dudde over at https://github.com/random-dudde/retina/pull/1 are poking at the [commit-message](https://github.com/microsoft/retina/blob/30a128b985bc99fc8686ef21afa1cc7358dc7dfd/.github/workflows/commit-message.yaml) Action trying to pull off a bash escape exploit.

The bash escape actually exists, due to the direct usage of the PR title:
```bash
commit_msg_header="${{ github.event.pull_request.title }}"
```

**However, this is not readily exploitable** because we require approval to run workflows on _all_ external contributions. A maintainer would need to approve the workflow, which makes it unlikely that any useful bash could be sneakily stuffed in to the title and executed.

Even getting workflow approval with a benign title and then updating it later is correctly handled by GH and requires a new maintainer approval:
image
preventing a TOCTOU malicious title swap.

With that all said...unlikely does not mean impossible, and even though it is not a zero-click attack, xz showed us that social engineering can be extremely effective.

This change removes the bash escape by staging the user-input in an [intermediate environment variable](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-intermediate-environment-variable) at the Job level. 

## Related Issue

If this pull request is related to any issue, please mention it here. Additionally, make sure that the issue is assigned to you before submitting this pull request.

## Checklist

- [x] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [x] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [x] I have correctly attributed the author(s) of the code.
- [x] I have tested the changes locally.
- [x] I have followed the project's style guidelines.
- [x] I have updated the documentation, if necessary.
- [x] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed

Please add any relevant screenshots or GIFs to showcase the changes made.

## Additional Notes

Add any additional notes or context about the pull request here.

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

### CODE REVIEW

File: .github/workflows/commit-message.yaml [hunk 1]
```diff
@@ -13,33 +13,28 @@ jobs:
...
-          echo 'example 3: "docs: update readme"'
-          echo
-          echo "Valid types are: $commit_msg_type_regex"
-          echo "For more details, see .github/workflows/commit-message.yaml"
-          exit 1
-        }
+          grep -qP "$full_regex" <<< "$TITLE" || {
+            echo "ERROR: Invalid commit message header. Please fix format of your PR title."
+            echo "Current value:"
+            echo "$TITLE"
```
Comment (vakalapa) [inline .github/workflows/commit-message.yaml:30]: Let's not print or echo the title ?

Comment (rbtr) [reply .github/workflows/commit-message.yaml]: sure, removed

**Review by timraymond — APPROVED** [review]

**Comment by robertprast** [comment]
Hey all, y'all hopped on this one quite quickly and nailed the impact assessment. Historically Ive reported similar findings on other Microsoft repos to the MSRC and was going about to fill a new report for this repo after looking more into any impact escalations (though no other vectors are jumping out to me). Overall solid stuff! Happy to provide any info on my end too

**Comment by rbtr** [comment]
Thanks @robertprast, if you find anything else do let us know 🙂 
MSRC reports are good and they will pass findings on to us pretty quickly. You can also [report directly](https://github.com/microsoft/retina/security/advisories/new) to immediately get our attention.

**Comment by random-dudde** [comment]
> Thanks @robertprast, if you find anything else do let us know 🙂 MSRC reports are good and they will pass findings on to us pretty quickly. You can also [report directly](https://github.com/microsoft/retina/security/advisories/new) to immediately get our attention.

I am running [poutine](https://github.com/boostsecurityio/poutine/releases) on public repos, and will let you know if there will be anything else.

**Pull request merged**